### PR TITLE
Changed URI.escape to CGI.escape

### PR DESCRIPTION
**Pull request by cameronlxu**
- In Ruby version 3.x, the `URI.escape` function becomes deprecated, causing issues that many users will run into.
- Changed the `URI.escape` function to become `CGI.escape` as it is more consistent with other projects such as [braintree_ruby](https://github.com/braintree/braintree_ruby).
- With this change, the space character now becomes a `+`, instead of previously being `%20`.
    - There are also other special/unsafe characters that are also encoded which is showcased below.

- Verified that the deprecation warning logs that appeared in _2.x_ do not appear in Ruby version _3.x_ when running tests:
Screen Shot 2022-06-14 at 11 51 59 AM

- Integration tested this change with the [PayPal Checkout-Ruby-SDK](https://github.com/paypal/Checkout-Ruby-SDK) and displayed no failures in unit tests.
   - Manually entered in test strings in the access token request body: https://github.com/paypal/Checkout-Ruby-SDK/blob/develop/lib/core/token_requests.rb#L7-L9
   - Temporarily added a print statement in the `FormEncoded.rb` file to view the results of the CGI.escape workings.
image

### CODE REVIEW

**Review by hlahlou-pp-dev — APPROVED** [review]

**Comment by josephschito** [comment]
This change is huge and LGTM!

**Review by crookedneighbor — CHANGES_REQUESTED** [review]
Sorry for the long delay on this. This looks very reasonable to me.

Can you add a CHANGELOG entry for it?

**Comment by cameronlxu** [comment]
@crookedneighbor Thank you! Just added version 1.0.2 with a line about the change.

File: CHANGELOG.md
Comment (crookedneighbor) [inline CHANGELOG.md:1]:
```suggestion
## unreleased
```

File: lib/paypalhttp/version.rb
Comment (crookedneighbor) [inline lib/paypalhttp/version.rb:1]:
```suggestion
VERSION = "1.0.1"
```

We'll do this at release time.

Comment (hlahlou-pp-dev) [reply lib/paypalhttp/version.rb:1]: Are you referring to this step within the release process? 

_gem push paypalhttp-1.X.X.gem_

In the previous deployments we have set the version manually while merging the PR before release: 

- https://github.com/paypal/paypalhttp_ruby/pull/1/files

- https://github.com/paypal/paypalhttp_ruby/pull/11/files

Do we want to approach this the same way or handle the version when we run the release steps?

Comment (crookedneighbor) [reply lib/paypalhttp/version.rb:1]: So 2 things that are important here.

First, we shouldn't set the version number until we actually do the release. Doing so within a PR doesn't allow us to do a release with a batch of changes from various PRs.

Second, this is _technically_ a breaking change to change the encoding from `%20` to `+`. It doesn't matter for the PayPal APIs, to my knowledge, but this module is for generic HTTP requests, so it's not guaranteed to be supported in all APIs. (most reasonable API systems will handle both, but we can't be sure about that). So even if we did set the version _in_ this PR, it'd need to be a 2.0.0 version.

File: lib/paypalhttp/serializers/form_encoded.rb
Comment (crookedneighbor) [inline lib/paypalhttp/serializers/form_encoded.rb:8]: Just curious, I see multiple PRs with this change with various implementations:

https://github.com/paypal/paypalhttp_ruby/pull/18
https://github.com/paypal/paypalhttp_ruby/pull/13
https://github.com/paypal/paypalhttp_ruby/pull/4

Is there a reason you chose this route over the others?

File: lib/paypalhttp/version.rb
Comment (hlahlou-pp-dev) [reply lib/paypalhttp/version.rb:1]: I completely agree that the version should be 2.0.0 . I will let them know to address these changes.

Well, the reason why we are changing the version number right now is that we plan to make a release right after having this change merged. (To address this ticket: https://engineering.paypalcorp.com/jira/browse/PPPLMER-108542) and so that Cameron can have his change go live before his internship ends. If you advise against making a release I understand, but there aren't typically many changes made to this repo.

Comment (crookedneighbor) [reply lib/paypalhttp/version.rb:1]: If you want to handle doing the release as a v2.0.0, then go for it.

File: lib/paypalhttp/serializers/form_encoded.rb
Comment (camlau-pp-dev) [reply lib/paypalhttp/serializers/form_encoded.rb:8]: The reason we decided to choose this implementation over the others is because it is more consistent with other maintained Paypal related repos. One repo in particular that we looked at was the [braintree_ruby repo](https://github.com/braintree/braintree_ruby). There are multiple instances where `CGI.escsape` is utilized, so we believed it was best to continue this consistency of using this specific function with repo as well.

Comment (crookedneighbor) [reply lib/paypalhttp/serializers/form_encoded.rb:8]: :+1: Thanks for the explanation

File: lib/paypalhttp/version.rb
Comment (hlahlou-pp-dev) [reply lib/paypalhttp/version.rb:1]: Ok thanks!

**Review by crookedneighbor — APPROVED** [review]

**Comment by camlau-pp-dev** [comment]
@crookedneighbor Apologies for the ping again, but since neither I nor Hakim have write access we are unable to merge this PR to start the release process. Thank you!

**Comment by fabn** [comment]
🎉

Will it be released on rubygems as well or should we point to github? 

Thanks for merging this.

**Comment by crookedneighbor** [comment]
@camlau-pp-dev and @HDLahlou are going to handle releasing it to Rubygems. They can update here when it's available.

**Comment by joshcass** [comment]
Looks like the payouts sdk will need a gemspec update or anyone using both gems will be unable to install `2.0.0`

https://github.com/paypal/Payouts-Ruby-SDK/blob/175338330b6af569ed0787a5a94ac0a584782f86/paypal-payouts-sdk.gemspec#L22

Also, thanks all for addressing this!

**Comment by fabn** [comment]
The same is true for PayPal checkout SDK https://github.com/paypal/Checkout-Ruby-SDK/blob/develop/paypal-checkout-sdk.gemspec#L22 which unfortunately has just been deprecated 🤦🏻‍♂️

Maybe you should reconsider version bump to 2.0.0 and go with 1.1 or something similar.

**Comment by camlau-pp-dev** [comment]
Hi everyone, we have just released the 2.0.0 version that can be seen here: https://rubygems.org/gems/paypalhttp/versions/2.0.0

**Comment by sweetppro** [comment]
will [Checkout-Ruby-SDK](https://github.com/paypal/Checkout-Ruby-SDK) be updated to use this?

**Comment by crookedneighbor** [comment]
@sweetppro No, as Checkout-Ruby-SDK is no longer receiving updates. You are welcome to fork it and update it yourself, or collaborate on a community fork of the SDK.

**Comment by magni-** [comment]
@crookedneighbor Will https://github.com/paypal/Payouts-Ruby-SDK be updated to use this? There's an open dependabot PR doing just that: https://github.com/paypal/Payouts-Ruby-SDK/pull/19

**Pull request merged**
